### PR TITLE
Format examples' titles and list

### LINKED ISSUE #1508: Improve titles of tutorials
## Issue Description
The titles of the tutorials could be improved. It is hard to understand right now the difference between tutorials. I propose to have some sort of classification, with a common structure. Like: {technique - topic - front end}, e.g., Using ZNE for H2 calculation with Cirq... or something similar.

### PR DESCRIPTION
**Pull request by nathanshammah**
## Description

Close #1508 by:
- Enabling a shorter format for titles in the gallery on the [Examples page](https://mitiq.readthedocs.io/en/latest/examples/examples.html) of the Mitiq documentation: All titles start with the acronym of the technique (ZNE or PEC). Generally then the frontend and backend (when relevant) are listed, additionally with more information on specifics, e.g., adding information on solving MaxCut "with QAOA". Generally tried to make titles more clear and consistent. 
- Reordering examples so that they are grouped per technique (PEC, ZNE), per example (same example with multiple languages / frontend are grouped), and generally alphabetically with regard to frontend (Braket, Cirq, Pyquil, Qiskit).

---

### License

- [ ] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.
- [ ] I added unit tests for new code.
- [ ] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [ ] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [ ] I [updated the documentation](../blob/master/docs/CONTRIBUTING_DOCS.md) where relevant.
- [ ] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

File: docs/source/examples/examples.myst [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
+ZNE with PyQuil: Parametric programs <pyquil_demo.ipynb>
+ZNE with PyQuil: Improving VQE <vqe-pyquil-demo.ipynb>
+ZNE with Cirq: Energy landscape of a variational circuit <simple-landscape-cirq.myst>
+ZNE with Qiskit: Energy landscape of a variational circuit <simple-landscape-qiskit.myst>
+ZNE with Cirq: Solving MaxCut with QAOA <maxcut-demo.myst>
+ZNE with Cirq: Hamiltonian simulation with Pauli gates<hamiltonians.md>
+ZNE with Cirq: Energy of molecular Hydrogen <molecular_hydrogen.myst>
+ZNE with PennyLane + Cirq: Energy of molecular Hydrogen <molecular_hydrogen_pennylane.myst>
+The Mitiq paper code <mitiq-paper/mitiq-paper-codeblocks>
+ZNE with OpenQASM: BQSKit compiler <bqskit.myst>
```
Comment (andreamari) [inline docs/source/examples/examples.myst:20]:
```suggestion
ZNE on a Rigetti backend with Braket: Mirror circuits <braket_mirror_circuit.md>
ZNE on IonQ backends with Braket <zne-braket-ionq.myst>
ZNE on IBM Quantum backends with Qiskit <ibmq-backends.myst>
ZNE on IBMQ backends with Pennylane <pennylane-ibmq-backends.myst>
ZNE on IBM Quantum backends with Cirq <cirq-ibmq-backends.myst>
ZNE with PyQuil: Parametric programs <pyquil_demo.ipynb>
ZNE with PyQuil: Improving VQE <vqe-pyquil-demo.ipynb>
ZNE with Cirq: Energy landscape of a variational circuit <simple-landscape-cirq.myst>
ZNE with Qiskit: Energy landscape of a variational circuit <simple-landscape-qiskit.myst>
ZNE with Cirq: Solving MaxCut with QAOA <maxcut-demo.myst>
ZNE with Cirq: Hamiltonian simulation with Pauli gates<hamiltonians.md>
ZNE with Cirq: Energy of molecular Hydrogen <molecular_hydrogen.myst>
ZNE with PennyLane + Cirq: Energy of molecular Hydrogen <molecular_hydrogen_pennylane.myst>
ZNE with OpenQASM: BQSKit compiler <bqskit.myst>
PEC on a Braket simulator: Mirror circuits <pec_tutorial.myst>
PEC with Cirq: Learning representations <learning-depolarizing-noise.myst>
The Mitiq paper code <mitiq-paper/mitiq-paper-codeblocks>
```

**Review by andreamari — APPROVED** [review]
I like the new titles! Much easier to read and to classify.
I left an **optional** suggestion for reordering in this way: ZNE-->PEC-->Other.

**Comment by nathanshammah** [comment]
Thanks @andreamari. My idea was that the PEC examples may be a bit "hidden" if they appear only after the so many ZNE examples. I also wanted to edit the list of examples in line with your comment, with some comments such as:
"Getting started with error mitigation:"
"Advanced use of error mitigation techniques:"
but couldn't find a way to do it. Any thoughts on this?

**Comment by nickdgardner** [comment]
I think this is a great structure for the titles. Likewise for the ordering scheme, with two thoughts:
1. Putting "The Mitiq paper code" last as Andrea has it seems natural to me.
2. For the ordering of ZNE vs. PEC, I'm not sure what I think (can see reasons for both). I have a slight preference for having PEC (and future techniques) come first, for the reason Nathan said and since it highlights the more complex techniques Mitiq supports which people might find exciting. Also if the order from bottom is "mitiq paper" as foundation, then ZNE, then others, there is for me a sort of nice pseudo-chronological and conceptual (increasing complexity) rationale to the ordering. But having ZNE first so simpler/introductory examples are at the top also makes sense.

@nathanshammah For the "Getting started..." vs "Advanced..." comments, do you mean like grouping the list into sections with those comments as heading?

**Review by natestemen — APPROVED** [review]
I have two minor concerns with this at the moment, but if you feel differently, I still like these changes, and hence am approving.

1. Renaming to put PEC/ZNE first makes it seem as though that's the important thing in the tutorial, when in reality our goal with mitiq is to make it as seamless as possible to use for a quantum programmer, and hence almost have them not even know it's there! I understand the point of doing this is so that a user scrolling through the tutorials will find them more scan-able, however, and I like that.
2. By making everything an acronym, the pages may be harder to find via those search for "zero noise extrapolation" and "probabilistic error cancellation" (searching ZNE and PEC don't give quantum results, unless postfixed with "quantum"). Looping back around to point 1, if someone is searching "variational quantum eigensolver tutorial" and comes across out tutorial currently titled "Variational quantum eigensolver improved with zero-noise extrapolation", it may seem more similar to what they are looking for than "ZNE with PyQuil: Improving VQE".

The first concern is especially small, but the second point is a little bigger. Do you think this could impact our SEO?

File: docs/source/examples/examples.myst [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
-Estimating the potential energy surface of molecular Hydrogen with ZNE and PennyLane + Cirq <molecular_hydrogen_pennylane.myst>
-Improving the accuracy of BQSKit compiled circuits with error mitigation <bqskit.myst>
-Learning quasiprobability representations with a depolarizing noise model <learning-depolarizing-noise.myst>
+PEC on a Braket simulator: Mirror circuits <pec_tutorial.myst>
+PEC with Cirq: Learning representations <learning-depolarizing-noise.myst>
+ZNE on a Rigetti backend with Braket: Mirror circuits <braket_mirror_circuit.md>
+ZNE on IonQ backends with Braket <zne-braket-ionq.myst>
+ZNE on IBM Quantum backends with Qiskit <ibmq-backends.myst>
+ZNE on IBMQ backends with Pennylane <pennylane-ibmq-backends.myst>
+ZNE on IBM Quantum backends with Cirq <cirq-ibmq-backends.myst>
```
[the remark this answers is not in the document]
Comment (nathanshammah) [reply docs/source/examples/examples.myst:10]: Thanks for spotting this. Actually the most recent format is IBM Quantum. https://twitter.com/jaygambetta/status/1581763882049150976

**Comment by nathanshammah** [comment]
Thanks all, I addressed all the comments I hope, adding a line of text at the beginning explicitly naming PEC and ZNE and reformatting the titles.

**Comment by nathanshammah** [comment]
@nickdgardner yes, probably that can be done with multiple galleries, but for the time being one seems enough.

**Review by natestemen — APPROVED** [review]
🚢

**Pull request merged**